src/briefcase/platforms/linux/appimage.py

```diff
@@ -252,34 +252,33 @@ def verify_tools(self):
         if self.tools.host_os != "Linux":
             raise BriefcaseCommandError("AppImages can only be executed on Linux.")
 
-    def run_app(self, app: AppConfig, **kwargs):
+    def run_app(self, app: AppConfig, test_mode: bool, **kwargs):
         """Start the application.
 
         :param app: The config object for the app
+        :param test_mode: Boolean; Is the app running in test mode?
         """
-        self.logger.info("Starting app...", prefix=app.app_name)
         try:
+            # Set up the log stream
+            kwargs = self._prepare_log_stream(app=app, test_mode=test_mode)
+
             # Start the app in a way that lets us stream the logs
             log_popen = self.tools.subprocess.Popen(
                 [os.fsdecode(self.binary_path(app))],
                 cwd=self.tools.home_path,
+                **kwargs,
                 stdout=subprocess.PIPE,
                 stderr=subprocess.STDOUT,
                 bufsize=1,
             )
 
             # Start streaming logs for the app.
-            self.logger.info("=" * 75)
-            self.tools.subprocess.stream_output(
-                app.app_name,
-                log_popen,
+            self._stream_app_logs(
+                app,
+                popen=log_popen,
+                test_mode=test_mode,
+                clean_output=False,
             )
-
-            # If the process didn't exit cleanly, raise an error.
-            if log_popen.returncode != 0:
-                raise BriefcaseCommandError(f"Problem running app {app.app_name}.")
-        except KeyboardInterrupt:
-            pass  # Catch CTRL-C to exit normally
         except OSError as e:
             raise BriefcaseCommandError(f"Unable to start app {app.app_name}.") from e
```

Comment:
I can see the logic you're going for here; I guess my counterargument would be that catching and re-raising a BriefcaseCommandError makes it clear that this is "expected unexpected behavior" - i.e., we've explicitly thought about this particular error condition, and know (at least to some level) what it means.

Reply:
This feels a bit like déjà vu....but I can't find the past discussion right now. All of Briefcase's uses of `subprocess` are exposed to `OSError` exceptions since they are all just wrappers on `Popen` and do not perform exception management.

So, ultimately, I'm not sure what's particularly different here that necessitates it. I feel like this may be more an artifact of macOS serving as the template for the other platforms; since on macOS, the `open` command returns as soon as it forks.....so, in that case, we're reporting the failure of the `open` command more so. That said, `OSError` actually being raised from `subprocess` here is probably quite unlikely.

Reply:
Agreed it's very unlikely; and I don't think it's especially necessary. It came up in my testing of AppImage specifically because I was futzing around with the actual appImages, and ended up with an existent, but 0 sized, non executable binary - something highly unlikely (impossible?) to actually happen in practice, but an edge case that *can* be caught, so I figured we might as well. 

Call this a "strong opinion, weakly held" on my part; It's there, it's not *hurting*, and on the off chance it occurs, we have a slightly nicer error message. If you feel particularly strongly to the contrary, I won't fight especially hard to defend it (here, or in the other backends where we're catching it). 

I'd be interested in @mhsmith's take on this, too.

Reply:
> we've explicitly thought about this particular error condition, and know (at least to some level) what it means.

I'm not sure that we know enough about what it means to justify suppressing it from the visible output. It could mean that the executable doesn't exist, or that it's corrupt as mentioned above, or that it's not executable. All of these would indicate a bug in Briefcase itself, so I think we should let the original exception through so you can see exactly has gone wrong without having to look in the log file. 

It would also make it as easy as possible for a user who receives one of those exceptions to search for it in the issue tracker, whereas the non-specific message would probably return a lot of irrelevant stuff.

Reply:
Ok - on that basis, I'll strip out the OSError handling on Popen calls.